### PR TITLE
[OPIK-2954] [BE] Model qwen/qwen2.5-vl-32b-instruct vision support by pattern matching

### PR DESCRIPTION
**Pull request by JetoPistola**
## Details

https://github.com/user-attachments/assets/44e21df8-105c-473c-bd7e-7349853aa41a

This PR implements vision support detection for Qwen VL models using pattern matching. The `ModelCapabilities.supportsVision()` method now checks against a hardcoded pattern (`.*qwen.*vl.*`) before falling back to JSON-based configuration. This approach handles cases where:
- Models are missing from the JSON configuration
- The JSON uses different naming conventions (e.g., openrouter/ vs deepinfra/ prefix)
- The JSON is not yet updated with new vision models

The pattern matching is case-insensitive and flexible, matching any model name containing "qwen" and "vl" in sequence.

Additionally, removed a duplicate entry for `qwen/qwen2.5-vl-32b-instruct` from the model prices JSON (the openrouter variant), keeping only the deepinfra variant to avoid confusion.

## Change checklist
- [x] User facing
- [x] Documentation update

## Issues
- OPIK-2954

## Testing
Added comprehensive test suite (`ModelCapabilitiesTest`) with 10 test cases covering:
- Pattern matching for qwen VL models (case insensitive)
- Known vision models (GPT-4, Claude, Gemini)
- Non-vision model detection
- Edge cases (null, blank, whitespace, colon suffixes)
- Provider prefix matching

All tests pass successfully.

## Documentation
No documentation changes required. This is an internal backend improvement to the model capabilities detection system.

### CODE REVIEW

File: apps/opik-backend/src/main/java/com/comet/opik/domain/llm/ModelCapabilities.java [hunk 1]
```diff
@@ -17,13 +17,38 @@
...
 @UtilityClass
 public class ModelCapabilities {
 
+    // Hardcoded vision-capable models or patterns that should support vision
+    // This list overrides the JSON configuration to handle cases where:
+    // - The JSON uses different naming conventions (e.g., openrouter/ prefix)
+    // - Models are missing from the JSON
+    // - The JSON is not yet updated with new vision models
+    private static final Set<Pattern> VISION_MODEL_PATTERNS = Set.of(
+            // Made pattern more flexible to match anywhere in the name
```
[the remark this answers is not in the document]
Comment (JetoPistola) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/llm/ModelCapabilities.java:32]: We might extend to new patterns, making this comment redundant

File: apps/opik-backend/src/main/java/com/comet/opik/domain/llm/ModelCapabilities.java [hunk 2]
```diff
@@ -17,13 +17,38 @@
...
 import java.util.Objects;
 import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Pattern;
 
 @UtilityClass
 public class ModelCapabilities {
 
+    // Hardcoded vision-capable models or patterns that should support vision
+    // This list overrides the JSON configuration to handle cases where:
```
[the remark this answers is not in the document]
Comment (JetoPistola) [reply apps/opik-backend/src/main/java/com/comet/opik/domain/llm/ModelCapabilities.java]: Applied

**Comment by CometActions** [comment]
✅ **Test environment is now available!**

### Access Information
- **URL:** https://pr-3945.dev.comet.com
- **Cluster:** comet-ml-development
- **Namespace:** pr-3945
- **Version:** 1.8.101.3945-merge-321
- **Application logs:** [View in Grafana](https://grafana.dev.comet.com/explore?schemaVersion=1&panes=%7B%22ai4%22%3A%7B%22datasource%22%3A%22P8E80F9AEF21F6940%22%2C%22queries%22%3A%5B%7B%22refId%22%3A%22A%22%2C%22expr%22%3A%22%7Bnamespace%3D%5C%22pr-3945%5C%22%7D%22%2C%22queryType%22%3A%22range%22%2C%22datasource%22%3A%7B%22type%22%3A%22loki%22%2C%22uid%22%3A%22P8E80F9AEF21F6940%22%7D%2C%22editorMode%22%3A%22builder%22%2C%22direction%22%3A%22forward%22%7D%5D%2C%22range%22%3A%7B%22from%22%3A%22now-1h%22%2C%22to%22%3A%22now%22%7D%7D%7D&orgId=1)

The deployment has completed successfully and the version has been verified.
